Issue/10207 pending upload labels

malinajirka opened the pull request:
Fixes #10207 

**This PR should be merged into `master-auto-upload` after https://github.com/wordpress-mobile/WordPress-Android/pull/10319 is merged.**

This PR updates labels on PostList items and SnackBar messages when the user confirms publishing/scheduling/saving/.. local changes in offline.

To test:
I'd consider testing this in [this](https://github.com/wordpress-mobile/WordPress-Android/pull/10343) PR.

Update release notes:

- The release-notes were updated in https://github.com/wordpress-mobile/WordPress-Android/pull/10319

shiki commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -261,6 +261,10 @@
     <string name="post_list_toggle_item_layout_cards_view">Switch to cards view</string>
     <string name="post_list_toggle_item_layout_list_view">Switch to list view</string>
+    <string name="post_waiting_for_connection_publish">Post will be published next time you device is online</string>
+    <string name="post_waiting_for_connection_pending">Post will be submitted for review next time your device is online</string>
```
This wasn't in the mockup but I like it. I updated the test scenarios to use this message. 👍

malinajirka replied:
Ohh I missed it was missing in the test scenarios! It's already approved text from https://github.com/wordpress-mobile/WordPress-Android/issues/10224.

malinajirka commented on WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt:
```diff
@@ -198,4 +178,41 @@ class UploadStarter @Inject constructor(
         }
     }
+
+    public fun getAutoUploadAction(post: PostModel, site: SiteModel): PostUploadAction {
```
💯 agree with this! My mind was telling me that it doesn't make sense and that I should move it. I was just lazy I guess :guilty:.

malinajirka replied:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10355

Unit test are tbd

maxme commented on WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java:
```diff
@@ -262,12 +262,20 @@ public static void publishPost(Activity activity, final PostModel post, SiteMode
+     * returns true if the user has permission to publish the post - assumed to be true for
+     * dot.org sites because we can't retrieve their capabilities
+     */
+    public static boolean userCanPublish(SiteModel site) {
```
I guess we should use this in the post list, maybe there https://github.com/wordpress-mobile/WordPress-Android/blob/2790aa65e4b6cac76113ba3c7992c942e29e01fa/WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt#L360 since we can't trust `hasCapabilityPublishPosts` for self hosted sites.

maxme replied:
BTW I filled an issue about "Submit"/"Publish" button on the post list for self hosted https://github.com/wordpress-mobile/WordPress-Android/issues/10354 we should maybe do a follow up PR for that one.

malinajirka replied:
Good find! I'll fix both issues in the PR for the issue.

malinajirka replied:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10356

malinajirka commented:
Thanks for the review @shiki !!! Great finds!

>The status label does not expand to accommodate the long message for private posts.

Fixed in 50260fb8f59d786b01d8294dbcbbb00975294884.

>This is probably out of the scope. I vaguely recall noticing this some time ago. If the user does not have Publish access, the Submit button assumes otherwise and shows the incorrect status message.

Fixed in b24db4c5d8bad001c7007db63b4d33192a8f6a9e. The issue was that we weren't updating the post status to Pending and we let the server to set the correct status.

maxme commented:
I'm not sure in when this was introduced, but I just noticed something weird on the post list for self hosted sites, here is a [demo video](https://bia.is/s/U9fr/local-draft-selfhosted-sites.webm)

- In airplane mode
- On a self hosted, create a new draft
- Tap the back button
- Check the post list, notice the local draft is shown on Published posts and on Draft posts

malinajirka commented:
>'m not sure in when this was introduced, but I just noticed something weird on the post list for self hosted sites, here is a demo video

I've created an issue https://github.com/wordpress-mobile/WordPress-Android/issues/10359. The bug is present even on `develop` so I moved it to `Post List Improvements` and `Groundskeeping` projects.

maxme commented:
:shipit: to the master branch
